airflow/models/dag.py

```diff
@@ -3902,6 +3901,16 @@ def get_dataset_triggered_next_run_info(self, *, session=NEW_SESSION) -> dict[st
             return None
         return get_dataset_triggered_next_run_info([self.dag_id], session=session)[self.dag_id]
 
+    @staticmethod
+    def get_dataset_expression(obj: DAG) -> Any | None:
+        """Deserialize the dataset_triggers structure from DAG into a dataset_expression."""
+        from airflow.serialization.serialized_objects import BaseSerialization  # avoid circular import
+
+        dataset_triggers = obj.dataset_triggers
```

Comment:
`dataset_triggers` is on `DAG`, not `DagModel`. I would make this a `@property` on `DAG`, and add a field on `DagModel` to store the same value by calling the property when a DagModel is created from a DAG.

Reply:
I have changed it